### PR TITLE
Update patch models for mongocluster

### PR DESCRIPTION
**Pull request by welovej**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

File: specification/mongocluster/DocumentDB.MongoCluster.Management/MongoCluster.tsp [hunk 1]
```diff
@@ -33,13 +34,12 @@ interface MongoClusters {
...
   /** Updates an existing mongo cluster. The request body can contain one to many of the properties present in the normal mongo cluster definition. */
-  update is ArmCustomPatchAsync<
-    MongoCluster,
-    Azure.ResourceManager.Foundations.ResourceUpdateModel<
-      MongoCluster,
-      MongoClusterProperties
-    >
-  >;
+  @sharedRoute
+  updateOld is ArmResourcePatchAsync<MongoCluster, MongoClusterProperties>;
```
Comment (live1206) [inline specification/mongocluster/DocumentDB.MongoCluster.Management/MongoCluster.tsp:38]: since we are updating the current version, no need to introduce updateOld

File: specification/mongocluster/resource-manager/Microsoft.DocumentDB/preview/2024-03-01-preview/examples/MongoClusters_UpdateOld.json [hunk 2]
```diff
@@ -1,19 +1,26 @@
 {
-  "operationId": "MongoClusters_Update",
-  "title": "Updates the disk size on a Mongo Cluster resource.",
+  "operationId": "MongoClusters_UpdateOld",
```
Comment (live1206) [inline specification/mongocluster/resource-manager/Microsoft.DocumentDB/preview/2024-03-01-preview/examples/MongoClusters_UpdateOld.json:2]: revert this since the operationId remains

File: specification/mongocluster/DocumentDB.MongoCluster.Management/MongoCluster.tsp (hunk 1 above)
Comment (welovej) [reply specification/mongocluster/DocumentDB.MongoCluster.Management/MongoCluster.tsp]: fixed

File: specification/mongocluster/DocumentDB.MongoCluster.Management/MongoCluster.tsp [hunk 3]
```diff
@@ -6,6 +6,7 @@ using TypeSpec.Rest;
 using Azure.ResourceManager;
 using Azure.ResourceManager.Private;
 using Azure.ClientGenerator.Core;
+using TypeSpec.Versioning;
```
Comment (live1206) [inline specification/mongocluster/DocumentDB.MongoCluster.Management/MongoCluster.tsp:9]: this is not needed anymore

File: specification/mongocluster/DocumentDB.MongoCluster.Management/client.tsp [hunk 4]
```diff
@@ -2,5 +2,22 @@ import "./main.tsp";
 import "@azure-tools/typespec-client-generator-core";
 
 using Azure.ClientGenerator.Core;
+using Azure.ResourceManager.CommonTypes;
+
+namespace Microsoft.DocumentDB;
```
Comment (live1206) [inline specification/mongocluster/DocumentDB.MongoCluster.Management/client.tsp:7]: should be `using Microsoft.DocumentDB`

File: specification/mongocluster/DocumentDB.MongoCluster.Management/tspconfig.yaml [hunk 5]
```diff
@@ -8,9 +8,6 @@ options:
     use-read-only-status-schema: true
     emitter-output-dir: "{project-root}/.."
     azure-resource-provider-folder: "resource-manager"
-    emit-common-types-schema: "never"
```
Comment (live1206) [inline specification/mongocluster/DocumentDB.MongoCluster.Management/tspconfig.yaml:11]: why are these removed?

File: specification/mongocluster/DocumentDB.MongoCluster.Management/MongoCluster.tsp (hunk 3 above)
Comment (welovej) [reply specification/mongocluster/DocumentDB.MongoCluster.Management/MongoCluster.tsp:9]: fixed

File: specification/mongocluster/DocumentDB.MongoCluster.Management/client.tsp (hunk 4 above)
Comment (welovej) [reply specification/mongocluster/DocumentDB.MongoCluster.Management/client.tsp:7]: fixed

File: specification/mongocluster/DocumentDB.MongoCluster.Management/tspconfig.yaml (hunk 5 above)
Comment (welovej) [reply specification/mongocluster/DocumentDB.MongoCluster.Management/tspconfig.yaml:11]: Misoperation

File: specification/mongocluster/DocumentDB.MongoCluster.Management/client.tsp [hunk 6]
```diff
@@ -2,5 +2,17 @@ import "./main.tsp";
 import "@azure-tools/typespec-client-generator-core";
 
 using Azure.ClientGenerator.Core;
+using Azure.ResourceManager.CommonTypes;
+using Microsoft.DocumentDB;
 
 @@clientName(Microsoft.DocumentDB, "MongoClusterMgmtClient", "python");
+@@access(Operations.list, Access.internal, "csharp");
+@@clientName(CheckNameAvailabilityResponse,
+  "CheckNameAvailabilityResult",
```
Comment (ArthurMa1978) [inline specification/mongocluster/DocumentDB.MongoCluster.Management/client.tsp:11]: MogoClusterNameAvailabilityResult

File: specification/mongocluster/DocumentDB.MongoCluster.Management/client.tsp [hunk 7]
```diff
@@ -2,5 +2,17 @@ import "./main.tsp";
...
+  "csharp"
+);
+@@clientName(MongoClusterRestoreParameters,
+  "MongoClusterRestoreContent",
+  "csharp"
+);
+@@clientName(CheckNameAvailabilityRequest.type, "ResourceType", "csharp");
```
Comment (ArthurMa1978) [inline specification/mongocluster/DocumentDB.MongoCluster.Management/client.tsp:18]: The `CheckNameAvailabilityRequest` should be renamed to `MogoClusterNameAvailabilityContent`

Comment (ArthurMa1978) [reply specification/mongocluster/DocumentDB.MongoCluster.Management/client.tsp:18]: And the operation should also be renamed to `CheckMongoClusterNameAvailability`

File: specification/mongocluster/DocumentDB.MongoCluster.Management/MongoCluster.tsp [hunk 8]
```diff
@@ -34,13 +34,9 @@ interface MongoClusters {
...
   /** Updates an existing mongo cluster. The request body can contain one to many of the properties present in the normal mongo cluster definition. */
-  update is ArmCustomPatchAsync<
-    MongoCluster,
-    Azure.ResourceManager.Foundations.ResourceUpdateModel<
-      MongoCluster,
-      MongoClusterProperties
-    >
-  >;
+  @parameterVisibility
+  update is ArmCustomPatchAsync<MongoCluster, MongoCluster>;
```
Comment (ArthurMa1978) [inline specification/mongocluster/DocumentDB.MongoCluster.Management/MongoCluster.tsp:38]: Just add the `@parameterVisibility` decorate, don't change the operation

File: specification/mongocluster/DocumentDB.MongoCluster.Management/examples/2024-03-01-preview/MongoClusters_PatchDiskSize.json [hunk 9]
```diff
@@ -1,60 +0,0 @@
-{
-  "operationId": "MongoClusters_Update",
```
Comment (ArthurMa1978) [inline specification/mongocluster/DocumentDB.MongoCluster.Management/examples/2024-03-01-preview/MongoClusters_PatchDiskSize.json:2]: Why do you remove the example?
If there are errors in the example, you can fix it or skip it in codegen

**Review by ArthurMa1978 — APPROVED** [review]
